docs/src/examples/components/Popup/Variations/PopupExamplePosition.shorthand.tsx

```diff
@@ -1,145 +1,67 @@
+  { position: 'after', align: 'bottom', icon: 'arrow circle right', padding: '18px 5px 5px 42px' },
+]
 
 const PopupExamplePosition = () => (
   <Grid columns="repeat(3, 30px)" variables={{ padding: '30px', gridGap: '80px' }}>
-    <Popup
-      align="start"
-      position="above"
-      trigger={triggers[0]}
-      content={
-        <p>
-          The popup is rendered above the trigger<br />aligned to the start.
-        </p>
-      }
-    />
-    <Popup
-      align="center"
-      position="above"
-      trigger={triggers[1]}
-      content={
-        <p>
-          The popup is rendered above the trigger<br />aligned to the center.
-        </p>
-      }
-    />
-    <Popup
-      align="end"
-      position="above"
-      trigger={triggers[2]}
-      content={
-        <p>
-          The popup is rendered above the trigger<br />aligned to the end.
-        </p>
-      }
-    />
-    <Popup
-      align="start"
-      position="below"
-      trigger={triggers[3]}
-      content={
-        <p>
-          The popup is rendered below the trigger<br />aligned to the start.
-        </p>
-      }
-    />
-    <Popup
-      align="center"
-      position="below"
-      trigger={triggers[4]}
-      content={
-        <p>
-          The popup is rendered below the trigger<br />aligned to the center.
-        </p>
-      }
-    />
-    <Popup
-      align="end"
-      position="below"
-      trigger={triggers[5]}
-      content={
-        <p>
-          The popup is rendered below the trigger<br />aligned to the end.
-        </p>
-      }
-    />
-    <Popup
-      align="top"
-      position="before"
-      trigger={triggers[6]}
-      content={
-        <p>
-          The popup is rendered before the trigger<br />aligned to the top.
-        </p>
-      }
-    />
-    <Popup
-      align="center"
-      position="before"
-      trigger={triggers[7]}
-      content={
-        <p>
-          The popup is rendered before the trigger<br />aligned to the center.
-        </p>
-      }
-    />
-    <Popup
-      align="bottom"
-      position="before"
-      trigger={triggers[8]}
-      content={
-        <p>
-          The popup is rendered before the trigger<br />aligned to the bottom.
-        </p>
-      }
-    />
-    <Popup
-      align="top"
-      position="after"
-      trigger={triggers[9]}
-      content={
-        <p>
-          The popup is rendered after the trigger<br />aligned to the top.
-        </p>
-      }
-    />
-    <Popup
-      align="center"
-      position="after"
-      trigger={triggers[10]}
-      content={
-        <p>
-          The popup is rendered after the trigger<br />aligned to the center.
-        </p>
-      }
-    />
-    <Popup
-      align="bottom"
-      position="after"
-      trigger={triggers[11]}
-      content={
-        <p>
-          The popup is rendered after the trigger<br />aligned to the bottom.
-        </p>
-      }
-    />
+    {triggers.map(({ position, align, icon, padding }) => (
```

Comment:
@levithomason thought we want to make these kind of examples explicit; at the same time we want to keep them short; what should be the approach here?

Reply:
I would rather follow the approach you've initially introduced for _these_ examples - as it seems the most reasonable one to me. Otherwise we will end up with extremely verbose code (once again, for this particular case) that will be very hard to grasp by the client. So, this is the reason I've preferred to use `map` and encapsulate not-changing parts